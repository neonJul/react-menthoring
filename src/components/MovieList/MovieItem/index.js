import React from 'react';

class MovieItem extends React.Component {
    render() {
        return (
            <div className="movie-item">
                <img src={this.props.img}/>
                <div className="movie-info">
                    <p className="movie-name">{this.props.name}</p>
                    <span className="movie-year">{this.props.year}</span>
                </div>
                <span>{this.props.genre}</span>
            </div>
        );
    }
}

export default MovieItem;