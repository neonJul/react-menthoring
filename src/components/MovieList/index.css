.info-panel {
    background-color: lightgray;
    padding: 10px 30px;
    display: flex;
    justify-content: space-between; 
}

.movie-list{
    padding: 30px;
    display: flex;
    flex-wrap: wrap;
}

.movie-item {
    width: 30%;
    margin: 10px;
}

.movie-item img {
    width: 100%;
    margin-bottom: 5px;
}

.movie-info {
    display: flex;
    justify-content: space-between;
}

.movie-name {
    font-weight: bold;
    text-transform: uppercase; 
}

.movie-year {
    align-self: flex-start;
    border: 1px solid lightgray;
    padding: 0 10px;
}