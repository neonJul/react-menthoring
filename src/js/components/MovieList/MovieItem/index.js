import React from 'react';

class MovieItem extends React.Component {
    render() {
        return (
            <div className="movie-item">
                <img src={require('./panda.jpg')}/>
                <div className="movie-info">
                    <p className="movie-name">Movie Name</p>
                    <span className="movie-year">year</span>
                </div>
                <span>genre</span>
            </div>
        );
    }
}

export default MovieItem;