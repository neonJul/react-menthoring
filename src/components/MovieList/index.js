import React from 'react';
import './index.css';
import Button from '../Shared/Button';
import MovieItem from './MovieItem';
import panda from '../../static/img/panda.jpg';

class MovieList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            sortByDate: true,
            sortByRating: false,
        }
    }

    sortByDate = () => {
        this.setState((prevState) => {
            return { 
                sortByDate: !prevState.sortByDate,
                sortByRating: !prevState.sortByRating
            }
        });
    }

    sortByRating = () => {
        this.setState((prevState) => {
            return { 
                sortByDate: !prevState.sortByDate,
                sortByRating: !prevState.sortByRating
            }
        });
    }

    render() {
        return (
            <React.Fragment>
                <div className="info-panel">
                    <span>xxx movies found</span>
                    <div className="sort-films">
                        <span>sort by</span>
                        <Button 
                            name="release date"
                            className={this.state.sortByDate ? "clicked-button" : "unclicked-button"}
                            onClick={this.sortByDate}
                        />
                        <Button 
                            name="rating"
                            className={this.state.sortByRating ? "clicked-button" : "unclicked-button"}
                            onClick={this.sortByRating}
                        />
                    </div>
                </div>
                <div className="movie-list">
                    <MovieItem name="Movie Name" img={panda} year="1996" genre="drama"/>
                    <MovieItem name="Movie Name" img={panda} year="1996" genre="genre"/>
                    <MovieItem name="Movie Name" img={panda} year="1996" genre="genre"/>
                    <MovieItem name="Movie Name" img={panda} year="1996" genre="genre"/>
                    <MovieItem name="Movie Name" img={panda} year="1996" genre="genre"/>
                </div>
            </React.Fragment>
        )
    }
};

export default MovieList;