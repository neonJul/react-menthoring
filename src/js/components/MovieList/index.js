import React from 'react';
import './index.css';
import Button from '../Shared/Button';
import MovieItem from './MovieItem';

class MovieList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            sortByDate: true,
            sortByRating: false,
        }
    }

    sortByDate = () => {
        this.setState((prevState) => {
            return { 
                sortByDate: !prevState.sortByDate,
                sortByRating: !prevState.sortByRating
            }
        });
    }

    sortByRating = () => {
        this.setState((prevState) => {
            return { 
                sortByDate: !prevState.sortByDate,
                sortByRating: !prevState.sortByRating
            }
        });
    }

    render() {
        return (
            <React.Fragment>
                <div className="info-panel">
                    <span>xxx movies found</span>
                    <div className="sort-films">
                        <span>sort by</span>
                        <Button 
                            name="release date"
                            className={this.state.sortByDate ? "clicked-button" : "unclicked-button"}
                            onClick={this.sortByDate}
                        />
                        <Button 
                            name="rating"
                            className={this.state.sortByRating ? "clicked-button" : "unclicked-button"}
                            onClick={this.sortByRating}
                        />
                    </div>
                </div>
                <div className="movie-list">
                    <MovieItem/>
                    <MovieItem/>
                    <MovieItem/>
                    <MovieItem/>
                    <MovieItem/>
                    <MovieItem/>
                    <MovieItem/>
                </div>
            </React.Fragment>
        )
    }
};

export default MovieList;