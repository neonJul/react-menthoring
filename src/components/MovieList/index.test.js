import React from 'react';
import renderer from 'react-test-renderer';

import MovieList from './index';

it('renders correctly', () => {
  const tree = renderer.create(<MovieList />).toJSON();
  expect(tree).toMatchSnapshot();
});