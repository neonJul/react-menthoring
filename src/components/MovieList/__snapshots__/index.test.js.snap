// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="info-panel"
  >
    <span>
      xxx movies found
    </span>
    <div
      className="sort-films"
    >
      <span>
        sort by
      </span>
      <button
        className="clicked-button button"
        onClick={[Function]}
      >
        release date
      </button>
      <button
        className="unclicked-button button"
        onClick={[Function]}
      >
        rating
      </button>
    </div>
  </div>,
  <div
    className="movie-list"
  >
    <div
      className="movie-item"
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movie-info"
      >
        <p
          className="movie-name"
        >
          Movie Name
        </p>
        <span
          className="movie-year"
        >
          year
        </span>
      </div>
      <span>
        genre
      </span>
    </div>
    <div
      className="movie-item"
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movie-info"
      >
        <p
          className="movie-name"
        >
          Movie Name
        </p>
        <span
          className="movie-year"
        >
          year
        </span>
      </div>
      <span>
        genre
      </span>
    </div>
    <div
      className="movie-item"
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movie-info"
      >
        <p
          className="movie-name"
        >
          Movie Name
        </p>
        <span
          className="movie-year"
        >
          year
        </span>
      </div>
      <span>
        genre
      </span>
    </div>
    <div
      className="movie-item"
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movie-info"
      >
        <p
          className="movie-name"
        >
          Movie Name
        </p>
        <span
          className="movie-year"
        >
          year
        </span>
      </div>
      <span>
        genre
      </span>
    </div>
    <div
      className="movie-item"
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movie-info"
      >
        <p
          className="movie-name"
        >
          Movie Name
        </p>
        <span
          className="movie-year"
        >
          year
        </span>
      </div>
      <span>
        genre
      </span>
    </div>
    <div
      className="movie-item"
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movie-info"
      >
        <p
          className="movie-name"
        >
          Movie Name
        </p>
        <span
          className="movie-year"
        >
          year
        </span>
      </div>
      <span>
        genre
      </span>
    </div>
    <div
      className="movie-item"
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movie-info"
      >
        <p
          className="movie-name"
        >
          Movie Name
        </p>
        <span
          className="movie-year"
        >
          year
        </span>
      </div>
      <span>
        genre
      </span>
    </div>
  </div>,
]
`;
